### PR TITLE
Refactor TaskEditorCommentPart, CommentGroupViewer, CommentViewer #266

### PR DESCRIPTION
**Pull request by BeckerFrank**
I rework the classes and now most fields are private again. I also introduce some new protected Methods so that we now can subclass TaskEditorCommentPart.

### CODE REVIEW

**Comment by ruspl-afed** [comment]
I need to have close look at it after finalizing #282

File: mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java [hunk 1]
```diff
@@ -669,6 +713,16 @@ protected void addActionsToToolbarTitle(ToolBarManager toolBarManager, TaskComme
 			CommentViewer commentViewer) {
 	}
 
+	protected void setTitleImage(ImageHyperlink expandCommentHyperlink, TaskComment taskComment,
```
Comment (ruspl-afed) [inline mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java:716]: may be a short comment on this method why it is empty? in fact this is a kind of internal API for customization, but empty method looks confusing a bit

File: mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java [hunk 2]
```diff
@@ -669,6 +713,16 @@ protected void addActionsToToolbarTitle(ToolBarManager toolBarManager, TaskComme
...
+			CommentViewer commentViewer) {
+
+	}
+
+	protected void setUserImage(UserAttributeEditor userImage, TaskAttribute commentAttribute,
```
Comment (ruspl-afed) [inline mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java:721]: may be a short comment on this method why it is empty? in fact this is a kind of internal API for customization, but empty method looks confusing a bit

File: mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java [hunk 3]
```diff
@@ -876,6 +930,10 @@ public List<CommentGroupViewer> getCommentGroupViewers() {
 		return commentGroupViewers;
 	}
 
+	protected CommentGroupViewer createCommentGroupViewer(List<CommentGroup> commentGroups, int i) {
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java]: Done

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/GitlabTaskEditorCommentPart.java [hunk 4]
```diff
@@ -15,376 +15,161 @@
...
     }
 
-    @Override
-    public List<CommentGroupViewer> getCommentGroupViewers() {
-	if (commentGroupViewers != null) {
-	    return commentGroupViewers;
-	}
+//    public GitlabTaskEditorCommentPart() {
+//	super();
+//	setPartName("Activities");
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/GitlabTaskEditorCommentPart.java]: Sorry this was left from an other Test. Maybe we split Comments ans System Events for the Gitlab Connector.

File: mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java (hunk 1 above)
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java:716]: Done. This is to allow subcallse to change the Default Image

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/GitlabTaskEditorCommentPart.java [hunk 5]
```diff
@@ -15,376 +15,161 @@
...
 	}
-
     }
 
-    @Override
-    public List<CommentGroupViewer> getCommentGroupViewers() {
-	if (commentGroupViewers != null) {
-	    return commentGroupViewers;
-	}
+//    public GitlabTaskEditorCommentPart() {
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/GitlabTaskEditorCommentPart.java]: done

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/GitlabTaskEditorCommentPart.java [hunk 6]
```diff
@@ -15,376 +15,161 @@
...
+    protected void setUserImage(UserAttributeEditor userImage, TaskAttribute commentAttribute,
 	    CommentViewer commentViewer) {
-	ReplyToCommentAction replyAction = new ReplyToCommentActionWithMenu(commentViewer);
-	replyAction.setImageDescriptor(TasksUiImages.COMMENT_REPLY_SMALL);
-	toolBarManager.add(replyAction);
+	String commentAuthor = getTaskData().getAttributeMapper().mapToRepositoryKey(commentAttribute,
+		TaskAttribute.COMMENT_AUTHOR);
+
+	TaskAttribute userImageAttribute = commentAttribute.getAttribute(commentAuthor);
+	TaskAttribute avatar_url = userImageAttribute.getAttribute("avatar_url");
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/GitlabTaskEditorCommentPart.java]: Done variable renamed

File: mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java [hunk 7]
```diff
@@ -553,6 +580,23 @@ private boolean hasTextControl() {
 					&& !commentTextEditor.getControl().isDisposed();
 		}
 
+		public void dispose() {
+			if (commentTextEditor != null && commentTextEditor.getControl() != null
```
[the remark this answers is not in the document]
Comment (ruspl-afed) [reply mylyn.tasks/org.eclipse.mylyn.tasks.ui/src/org/eclipse/mylyn/internal/tasks/ui/editors/TaskEditorCommentPart.java:584]: ok, may be later

**Review by ruspl-afed — APPROVED** [review]
Thank you for considering my comments.
LGTM

**Pull request merged**